server_test.go

```diff
@@ -575,6 +579,96 @@ func TestShutdownTCP(t *testing.T) {
 	}
 }
 
+func checkInProgressQueriesAtShutdownServer(t *testing.T, srv *Server, addr string, client *Client) {
```

Comment:
We need to figure this one out. I'm fine with merging this *without* this test and then having a thought. This will either be flaky or hard to debug when it fails.

Reply:
@miekg I can do that, but it does seem to be a little risky. It's very hard to debug at the moment, I really can't figure out why `TestInProgressQueriesAtShutdownTLS` is failing for go1.10 but not go1.11 and only on Travis. Without a solid test, I don't feel completely confident in this PR.